### PR TITLE
Service class refactoring

### PR DESCRIPTION
**Pull request by k9ert**
Some changes to the extension-framework:
* Specter now has a dictionary called `ext` which contains all the extensions. This might be a semi-good idea.
* As we now might have external extensions in separate repos and therefore directories like `src/cryptoadvance`, we need a more sophisticated method to detect, whether we're in the specter-desktop repo-folder.
* There is now a way to schedule events with the help of the `Flask-APScheduler` module. A scheduler instance will be created and passed to the `after_serverpy_init_app` callback. The dummyextension has some example code on how to use this.
* Now there is also an `ServiceUnencryptedStorage` class available in order to store service specific stuff unencrypted.
* Some bugfixes

### CODE REVIEW

**Comment by moneymanolis** [comment]
Opportunity to change Services to Plugins in this PR
image

File: src/cryptoadvance/specter/services/callbacks.py [hunk 1]
```diff
@@ -11,5 +11,6 @@
     
 """
 
-afterServiceManagerInit = "afterServiceManagerInit"
+afterServiceManagerInit = " afterServiceManagerInit"
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/services/callbacks.py:14]:
```suggestion
afterServiceManagerInit = "afterServiceManagerInit"
```

File: src/cryptoadvance/specter/services/service_encrypted_storage.py [hunk 2]
```diff
@@ -98,6 +99,14 @@ def get_service_data(self, service_id: str) -> dict:
         return service_data
 
 
+class ServiceUnencryptedStorage(ServiceEncryptedStorage):
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/services/service_encrypted_storage.py:102]: Is that really enough? No "encrypted=False" etc.

Comment (k9ert) [reply src/cryptoadvance/specter/services/service_encrypted_storage.py:102]: I've added some sanity-checking and a documentation for that class.

**Pull request merged**